src/vs/workbench/contrib/terminal/browser/terminalProfileService.ts

```diff
@@ -67,10 +68,14 @@ export class TerminalProfileService implements ITerminalProfileService {
 
 		this._webExtensionContributedProfileContextKey = TerminalContextKeys.webExtensionContributedProfile.bindTo(this._contextKeyService);
 		this._updateWebContextKey();
-		// Wait up to 5 seconds for profiles to be ready so it's assured that we know the actual
-		// default terminal before launching the first terminal. This isn't expected to ever take
-		// this long.
-		this._profilesReadyBarrier = new AutoOpenBarrier(20000);
+		this._profilesReadyPromise = this._remoteAgentService.getEnvironment()
+			.then(() => {
+				// Wait up to 20 seconds for profiles to be ready so it's assured that we know the actual
+				// default terminal before launching the first terminal. This isn't expected to ever take
+				// this long.
+				this._profilesReadyBarrier = new AutoOpenBarrier(20000);
+				return this._profilesReadyBarrier.wait().then(() => { });
+			});
```

Comment:
I lack the context why this barrier was needed before, maybe is not needed anymore and we can remove it? With this change we start the barrier after the remote have been completely resolved, before that vscode is in `Opening remote...` state waiting for the extension RemoteAuthorityResolver to finish

Reply:
We could maybe remove it, but I'm a little scared to tbh 🙂